src/logic/logic.ts

```diff
@@ -3,14 +3,34 @@
  * Utilities for working with program bytes.
  */
 
-const langspec = require('./langspec.json');
+import langspec from './langspec.json';
 
-let opcodes;
+/**
+ * Langspec Op Structure
+ */
+interface OpStructure {
+  Opcode: number;
+  Name: string;
+  Args?: string;
+  Returns?: string;
+  Cost: number;
+  Size: number;
+  ArgEnum?: string[];
+  ArgEnumTypes?: string;
+  Doc: string;
+  DocExtra?: string;
+  ImmediateNote?: string;
+  Groups: string[];
+}
+
+let opcodes: {
+  [key: number]: OpStructure;
+};
 
 const maxCost = 20000;
 const maxLength = 1000;
 
-function parseUvarint(array) {
+export function parseUvarint(array: Uint8Array) {
```

Comment:
Could you add a named return type of `[numberFound: number, size: number]` to this function?

Reply:
Good idea!